### PR TITLE
Create auth token management page

### PR DESCRIPTION
**Pull request by williamjallen**
Individual users can create authentication tokens but there is no way to disable individual authentication tokens system-wide.  The ability for administrators to disable a users's tokens may be desirable in some environments.

This PR adds an admin-only authentication token management page which allows administrators to find and delete any authentication token created on the system.

image

While making this PR, I did some file rearrangement to match Laravel's best practices.  In particular, I eliminated the PHP view class, since the Blade templates are supposed to serve the place of a view.  Having the directory `Http/Controllers/Views/...` was particularly confusing since there was also a `Resources/views` directory containing Blade templates.  I also renamed our abstract classes to include the word "abstract".  This naming scheme will make it easier to identify them as base classes to be extended since they are otherwise mixed in with the rest of the controllers.

I also began preliminary work on a Laravel-based API, which is currently hosted in the main routes file.  The proper Laravel way to do this is to have bearer tokens authenticate API requests, with routes remaining in a separate API routes file.  In the future, I will move over the API routes to use this scheme.

### CODE REVIEW

File: app/cdash/app/Model/Build.php [hunk 1]
```diff
@@ -2116,8 +2116,8 @@ public function UpdateParentTestNumbers($newFailed, $newNotRun, $newPassed)
                     'testpassed' => $numPassed,
                 ]);
 
-        // NOTE: as far as I can tell, build.testtimestatusfailed isn't used,
-        // so for now it isn't being updated for parent builds.
+            // NOTE: as far as I can tell, build.testtimestatusfailed isn't used,
+            // so for now it isn't being updated for parent builds.
```
Comment (williamjallen) [inline app/cdash/app/Model/Build.php:2120]: The linter made this change, but it doesn't seem to be directly related to any code changes I made.  Furthermore, I am unable to reproduce any linting failures locally, despite the CI failing [here](https://app.circleci.com/pipelines/github/Kitware/CDash/1015/workflows/caab9e83-be1e-4a69-aea8-36884845e3a2/jobs/4515?invite=true#step-104-348), with further changes to this file suggested.

Yet another weird issue with the style check, in a continuation of the issues described in https://github.com/Kitware/CDash/issues/1309.

Comment (williamjallen) [reply app/cdash/app/Model/Build.php]: Issue seems to have been resolved by reverting the changes to `Build.php`.  I am possibly running a different version of the style checker.  This type of issue would be fixed by pinning all of our dependencies, as discussed in https://github.com/Kitware/CDash/issues/1313.

**Comment by williamjallen** [comment]
@josephsnyder This PR is failing the CI with the same error you have noted.  The issue is Laravel's CSRF token handling--the old API doesn't use them, and the tests don't have a token to use when they make submissions.

You can set `APP_ENV=testing` to disable CSRF tokens for testing purposes, which is probably the best long-term solution here.  I attempted to set it in `docker/.env.testing`, but it didn't have any effect.  When I changed the default value of `APP_ENV` in `config/app.php` to `testing`, CSRF tokens were disabled and the tests passed.

I'm currently out of ideas when it comes to getting `APP_ENV` set appropriately in the CI so the tests can pass.  A few other places in the codebase use a hack to get the CSRF token each time, but that is not a good long-term solution as move towards an entirely Laravel-based API.

I'll rebase the PR and make the timestamp formatting changes you noted.

**Comment by williamjallen** [comment]
I have changed the default value of `APP_ENV` to `testing`, which disables CSRF token verification.  It is necessary for the tests to pass, and I think it is reasonable to assume that most non-production instances want to be able to run the tests locally.

I am open to other thoughts/comments on this, since it doesn't feel like a great solution.

**Review by josephsnyder — APPROVED** [review]
Looks good!

**Pull request merged**